allen-cell-animated/cell-feature-explorer
# fix: Full field toggle is no longer visible on datasets without full field

Problem
=======
Closes #174, "full field is still visible for datasets without it."

I've thrown in a couple other misc. changes but this is a small PR (<20 lines). Most of the line count comes from dependency changes.

*Estimated review size: tiny, <5 minutes*

Solution
========
- Passes an empty string if the `fovPath` is an empty string.
- Fixes formatting on the left toolbar in the viewport.
- Adds the `ViewerStateProvider` as per the new W3CV API.
- Upgrade web-3d-viewer to v2.9.2

## Type of change
Please delete options that are not relevant.

* Bug fix (non-breaking change which fixes an issue)

Steps to Verify:
----------------
1. Clone and run `npm run start`.
2. Open http://localhost:9002/?cellSelectedFor3D=439543&colorBy=cell-line&dataset=variance_v1&plotByOnX=cell-volume&plotByOnY=dna-volume&selectedPoint%5B0%5D=439543. You should see the full field toggle.
3. Open http://localhost:9002/?cellSelectedFor3D=F2492C26&colorBy=cell-age&dataset=cellsystems_live_v2021.1&plotByOnX=expert-score&plotByOnY=cos&selectedPoint%5B0%5D=F2492C26. You should not see the full field toggle.

Screenshots (optional):
-----------------------
Full field available:
![image](https://github.com/user-attachments/assets/025989fc-ecd8-4641-b5cc-5f6c836f97b3)

Full field not available:
![image](https://github.com/user-attachments/assets/2c28fc64-8e63-4e49-b381-269bf14057be)
